PythonAPI/carla/src/SensorData.cpp

```diff
@@ -300,29 +300,49 @@ static FakeImage ColorCodedFlow (
 
 template <typename T>
 static std::string SaveImageToDisk(T &self, std::string path, EColorConverter cc) {
+  std::string r;
   carla::PythonUtil::ReleaseGIL unlock;
   using namespace carla::image;
+  puts("ImageView::MakeView.");
   auto view = ImageView::MakeView(self);
+  puts("ImageIO::WriteView.");
+  try
+  {
   switch (cc) {
     case EColorConverter::Raw:
-      return ImageIO::WriteView(
+      r = ImageIO::WriteView(
           std::move(path),
           view);
+      break;
     case EColorConverter::Depth:
-      return ImageIO::WriteView(
+      r = ImageIO::WriteView(
           std::move(path),
           ImageView::MakeColorConvertedView(view, ColorConverter::Depth()));
+      break;
     case EColorConverter::LogarithmicDepth:
-      return ImageIO::WriteView(
+      r = ImageIO::WriteView(
           std::move(path),
           ImageView::MakeColorConvertedView(view, ColorConverter::LogarithmicDepth()));
+      break;
     case EColorConverter::CityScapesPalette:
-      return ImageIO::WriteView(
+      r = ImageIO::WriteView(
           std::move(path),
           ImageView::MakeColorConvertedView(view, ColorConverter::CityScapesPalette()));
+      break;
     default:
       throw std::invalid_argument("invalid color converter!");
   }
+  }
+  catch (std::exception& e)
+  {
+    puts(e.what());
+  }
+  catch (...)
+  {
+    puts("Exception thrown");
+  }
+  puts("Done.");
```

Comment:
This `puts("Done.")` will be called also when an exception was thrown.
Is this correct?